cirq-core/cirq/work/observable_measurement.py

```diff
@@ -463,49 +528,29 @@ def measure_grouped_settings(
             in `circuit`. The total sweep is the product of the circuit sweep
             with parameter settings for the single-qubit basis-change rotations.
         readout_calibrations: The result of `calibrate_readout_error`.
-        checkpoint: If set to True, save cumulative raw results at the end
-            of each iteration of the sampling loop. Load in these results
-            with `cirq.read_json`.
-        checkpoint_fn: The filename for the checkpoint file. If `checkpoint`
-            is set to True and this is not specified, a file in a temporary
-            directory will be used.
-        checkpoint_other_fn: The filename for another checkpoint file, which
-            contains the previous checkpoint. This lets us avoid losing data if
-            a failure occurs during checkpoint writing. If `checkpoint`
-            is set to True and this is not specified, a file in a temporary
-            directory will be used. If `checkpoint` is set to True and
-            `checkpoint_fn` is specified but this argument is *not* specified,
-            "{checkpoint_fn}.prev.json" will be used.
+        checkpoint: Options to set up optional checkpointing of intermediate
+            data for each iteration of the sampling loop. See the documentation
+            for `CheckpointFileOptions` for more. Load in these results with
+            `cirq.read_json`.
     """
     if readout_calibrations is not None and not readout_symmetrization:
         raise ValueError("Readout calibration only works if `readout_symmetrization` is enabled.")
 
-    checkpoint_fn, checkpoint_other_fn = _parse_checkpoint_options(
-        checkpoint=checkpoint, checkpoint_fn=checkpoint_fn, checkpoint_other_fn=checkpoint_other_fn
-    )
     qubits = sorted({q for ms in grouped_settings.keys() for q in ms.init_state.qubits})
     qubit_to_index = {q: i for i, q in enumerate(qubits)}
 
     needs_init_layer = _needs_init_layer(grouped_settings)
     measurement_param_circuit = _with_parameterized_layers(circuit, qubits, needs_init_layer)
-    grouped_settings = {
-        _pad_setting(max_setting, qubits): settings
```

Comment:
the call to `_pad_setting` has been moved to right near where it's actually needed in `_get_params_for_setting`. This was causing a subtle bug with the new test I added that uses a particularly bad grouper: it puts each observable in its own group. However, if you pad the keys in this bad grouping you start to get collisions. We'll keep the user's input `max_setting`s whenever we're using it as a key anywhere and just "pad" when we need the actual circuit parameters.